azure-pipelines.yml

```diff
@@ -0,0 +1,80 @@
+stages:
+ - stage: check
+   displayName: Compilation check
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+         name: cargo_check
+ # This represents the minimum Rust version supported.
+ # Tests are not run as tests may require newer versions of rust.
+ - stage: msrv
+   displayName: "Minimum supported Rust version"
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+        rust: 1.34.0
+ - stage: custom
+   displayName: "Special tests"
+   dependsOn: check
+   pool:
+     vmImage: ubuntu-16.04
+   jobs:
+     - job: custom_stable
+       displayName: "Stable custom tests"
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: stable
+         - bash: cd tracing/test-log-support && cargo test
+           displayName: "Test log support"
+         - bash: cd tracing/test_static_max_level_features && cargo test
+           displayName: "Test static max level features"
+     - job: custom_nightly
+       displayName: "Nightly custom tests"
+       continueOnError: true
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: nightly
+         - script: cargo test --features=doctest-readme --all
+           displayName: "Test README as doctest"
+         - bash: cd tracing-futures/test_std_future && cargo test
+           displayName: "Test std::future support for tracing-futures"
+ - stage: style
+   displayName: Style linting
+   dependsOn: ${{ format('{0}check', parameters.prefix) }}
+   jobs:
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt_beta
+        rust: beta
+        allow_fail: true
+    - template: azure/cargo-check.yml@templates
+      parameters:
+        name: warning_free
+        err_on_warn: true
+        allow_fail: true
+ - stage: test
+   displayName: Test suite
+   dependsOn: check
+   jobs:
+     - template: azure/tests.yml@templates
+ - stage: coverage
+   displayName: Code coverage
+   dependsOn: test
+   jobs:
+     - template: azure/coverage.yml@templates
+       parameters:
+         codecov_token: $(CODECOV_TOKEN_SECRET)
```

Comment:
I'm assuming this will currently fail as we don't have a codecov token set?

Reply:
It will actually just skip coverage if the token isn't set.